polars/polars-core/src/datatypes/dtype.rs

```diff
@@ -256,8 +257,13 @@ impl Display for DataType {
             DataType::Int64 => "i64",
             DataType::Float32 => "f32",
             DataType::Float64 => "f64",
-            #[cfg(feature = "dtype-i128")]
-            DataType::Decimal128(_) => "i128",
+            #[cfg(feature = "dtype-decimal")]
+            DataType::Decimal(prec, scale) => {
+                return f.write_str(&match prec {
+                    Some(prec) => format!("decimal[{}, {}]", prec, scale),
+                    None => format!("decimal[*, {}]", scale),
```

Comment:
I've thought about this too, and I don't really enjoy the look of it either :)
- scale is _way_ more important
- precision will often be just None
- if scale and precision were flipped, `precision=None` could become an optional kwarg from decimal datatype

So we could do...
- Something like `decimal[5]` (scale=5, prec=None) and `decimal[5(2)]` (scale=5, prec=2)
- But the above probably only makes sense if we swap prec/scale everywhere (DataType in Rust, Decimal datatype in Python, etc), otherwise it's even more confusing....

Reply:
Another middle-ground solution would be not flipping prec/scale, but printing it as `decimal[5]` if prec is None or something like `decimal{2}[5]` if prec=2. Idk, still a bit ugly and possibly confusing 🤔

Reply:
`decimal[5(2)]` LGTM as well.

An alternative for showing `scale`, `precision` in that order would be to use a dot in front of scale making it more explicit, e.g.: `decimal[.5,2]` Then various separators could be used, e.g.: `:` or `;` perhaps.

> But the above probably only makes sense if we swap precision/scale everywhere (DataType in Rust, Decimal datatype in Python, etc), otherwise it's even more confusing....

Yeah, good point, we'd have to flip it around everywhere. Probably not worth it. It would also mean it'd be reversed from Arrow's definition which can be somewhat confusing.

No strong opinion. Using `_` would probably pass as "pythonic" as well.

Reply:
@plaflamme Ok, changed it to `decimal[2]` and `decimal[.5,2]`.